### PR TITLE
Fix/jwt errorhandling

### PR DESCRIPTION
**Pull request by Tieske**
replaces #1247, see also #1244.

### CODE REVIEW

File: kong/plugins/jwt/handler.lua [hunk 1]
```diff
@@ -56,13 +56,20 @@ function JwtHandler:access(conf)
...
+    else
+      return responses.send_HTTP_UNAUTHORIZED("Unrecognizable token")
+    end
   end
-
+  
   -- Decode token to find out who the consumer is
   local jwt, err = jwt_decoder:new(token)
   if err then
+    -- TODO: is this an internal server error? the token is invalid/cannot be parsed so isn't this a user error to be reported?
```
Comment (Tieske) [inline kong/plugins/jwt/handler.lua]: @thibaultCha ^^

The possible errors [originate here](https://github.com/Mashape/kong/blob/74940cfbc43bd5eae25bdd86cbde664ae6294adb/kong/plugins/jwt/jwt_parser.lua#L99-L117)

They seem user errors to me, any specific reason they are set to fail with a 500?

Comment (thibaultCha) [inline kong/plugins/jwt/handler.lua]: Yeah those are user errors indeed.

File: kong/plugins/jwt/handler.lua [hunk 2]
```diff
@@ -56,13 +56,20 @@ function JwtHandler:access(conf)
     return responses.send_HTTP_INTERNAL_SERVER_ERROR(err)
   end
 
-  if not token then
-    return responses.send_HTTP_UNAUTHORIZED()
+  if type(token) ~= "string" then
+    if type(token) == "nil" then
```
Comment (thibaultCha) [inline kong/plugins/jwt/handler.lua]: `token == nil`?

**Comment by Tieske** [comment]
@thibaultCha both remarks updated

File: kong/plugins/jwt/api.lua [hunk 3]
```diff
@@ -11,11 +12,35 @@ return {
...
+          return helpers.responses.send_HTTP_BAD_REQUEST("No mandatory 'rsa_public_key'")
+       end
+
+      if self.params.algorithm == "RS256" and crypto.pkey.from_pem(self.params.rsa_public_key) == nil then
+        return helpers.responses.send_HTTP_BAD_REQUEST("'rsa_public_key' format is invalid")
+      end
+
+      if self.params.algorithm == "HS256" and self.params.secret == nil then
+        return helpers.responses.send_HTTP_BAD_REQUEST("No mandatory 'secret'")
+      end
```
Comment (thibaultCha) [inline kong/plugins/jwt/api.lua]: The API is not the right place to perform schema validation. Schema validation must be performed at the Model layer, aka in the `schema.lua` file.

Comment (Tieske) [inline kong/plugins/jwt/api.lua]: Thx, I see what you mean!

**Pull request merged**